zap/src/parser/grammar.lalrpop

```diff
@@ -160,28 +160,28 @@ Struct: SyntaxStruct<'input> = {
     <start:@L> "{" <fields:Comma<(<KeyIdentifier> ":" <Ty>)>> "}" <end:@R> => SyntaxStruct { start, fields, end },
 }
 
-IntRange: SyntaxRange<'input> = {
-    <start:@L> <kind:IntRangeKind> <end:@R> => SyntaxRange { start, kind, end },
+UnsignedNumTy: NumTy = {
```

Comment:
Defining a NumTy type in the grammar and using it as

```
<numty:NumTy> <r:("(" <NumRange> ")")?> => SyntaxTyKind::Num(numty, r),
```

in the TyKind & ranges' upper bounds should be a cleaner and more maintainable solution.